### ISSUE TITLE
Docs do not mention discontinuation of support for Python 3.7 in TF 2.11.1

### ISSUE THREAD

**Issue by varunshankar** [tensorflow/tensorflow#60176]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

tf 2.11.1

### Custom Code

No

### Python version

3.7

### Current Behaviour?

```shell
The release notes for TensorFlow 2.12.0 state that support for Python 3.7 has been discontinued. However, it seems that Python 3.7 support has also been removed from TensorFlow 2.11.1, despite this not being mentioned in the release notes or documentation. As a result, to maintain compatibility, we were forced to limit our application to using TensorFlow 2.11.0.
```

### Standalone code to reproduce the issue

```shell
NA
```

### Relevant log output

_No response_</details>

**Comment by sushreebarsa** [comment]
Hello @varunshankar! COuld you please have a look at [this](https://www.tensorflow.org/install/source_windows) page which explains well about the tested build configuration for different compatible versions. The compatible python version for TF v2.12. is 3.8 - 3.11  .
Thank you!

**Comment by thomdixon** [comment]
2.11.1 contains a critical security update that would be nice to have with the same version of Python which is compatible with 2.11 (i.e., Python 3.7). Can you please distribute 2.11.1 with Python 3.7 support?

**Comment by shaunakgodbole** [comment]
@sushreebarsa we expect that there would be a build for python 3.7 for TF 2.11.1 especially given the number of vulnerabilities that are resolved in that release. Can you please provide with a tentative timeline on when this will be resolved?

**Comment by nvbkdw** [comment]
Hi @sushreebarsa I wonder if it's intentional to discontinue support for Python 3.7? Is there a technical problem preventing 2.11.1 patch been built on Python 3.7? (we need this patch to deploy on Python 3.7, so want to find a resolution)

**Comment by camattin** [comment]
It looks like python 3.7 was removed from some of the automations back in December [here](https://github.com/tensorflow/tensorflow/commit/90d257096c7afcc499ef8a934e78929fb3b7531d). Perhaps this is release to the reason we're not seeing 2.11.1 artifacts for python 3.7?

FWIW, I was able to compile 2.11.1 with python 3.7 and successfully install the built wheel. This is without thorough testing, though, so I cannot say it's 100% functional.

**Comment by YONGARYZ** [comment]
Hi @sushreebarsa  
I'm also one of developers waiting your reply.
My team is deploying TF model using python 3.7 and we need to solve vulnerabilities mentioned above(ex. CVE-2023-25668). 
If TF 2.11.1 is not supporting python 3.7 any more, I need a plan B ASAP.
I look forward to hearing from you soon
Thanks in advance

**Comment by sushreebarsa** [comment]
@varunshankar @YONGARYZ Sorry for the late response!

Please have a look at the documentation here which clearly mentions the build configuration of TF version 2.11. For TF v2.11 , the supported version of python is 3.7- 3.10. That means there is no discontinuation of python 3.7. If you see any compatibility issues then please let us know.  Thank you!

**Comment by alcysec** [comment]
Hey @sushreebarsa. The issue is that the `tensorflow-cpu` version 2.11.1 for Python 3.7 isn't available on PyPi, which might give the impression that support for 3.7 has been discontinued.

```shell
$ docker run --platform=linux/amd64 --entrypoint /bin/bash python:3.7-slim -c "pip install tensorflow-cpu==2.11.1"
ERROR: Could not find a version that satisfies the requirement tensorflow-cpu==2.11.1 (from versions: 1.15.0, 2.1.0, 2.1.1, 2.1.2, 2.1.3, 2.1.4, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.4.0, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.9.0rc0, 2.9.0rc1, 2.9.0rc2, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.10.0rc0, 2.10.0rc1, 2.10.0rc2, 2.10.0rc3, 2.10.0, 2.10.1, 2.11.0rc0, 2.11.0rc1, 2.11.0rc2, 2.11.0)
ERROR: No matching distribution found for tensorflow-cpu==2.11.1
```

**Comment by YONGARYZ** [comment]
@sushreebarsa  I have a same issue like @alcysec.
As you mentioned, the documentation from Tensorflow, PyPi[(link)](https://pypi.org/project/tensorflow-cpu/2.11.1/) clearly mentions TF 2.11.1 supports python 3.7. But I can't install using pip. (I'm using an official image python:3.7.13-slim. )
The reason might be the @camattin's mention above.

**Comment by YONGARYZ** [comment]
@sushreebarsa 
I'm still looking forward to your reply, but the issue has been closed.

**Comment by sushreebarsa** [comment]
@YONGARYZ Reopening the issue as per the request. Thank you!

**Comment by YONGARYZ** [comment]
@sushreebarsa Thank you for reopening the issue.
Could you please make any comment about the issue?
As you mentioned in the previous comment ("For TF v2.11 , the supported version of python is 3.7- 3.10."), definitely there is a problem.
I need to make countermeasures for this issue. Therefore please let me know about whether you (or other contributors) are recognizing the issue. if then could you tell me brief plan for solving the issue? I need to make a decision in real quick.

**Comment by sachinprasadhs** [comment]
Hi, 

Thanks for reporting the issue, The usual practice in Tensorflow to support any Python version is, during the time of release if that Python version is within the support life cycle as defined here https://devguide.python.org/versions/.
As per the above link, during the patch release of `2.11.1` Python `3.7` was at `end-of-life `.

For any CVEs which needs the specific python version, you can follow out [Patching guidelines](https://github.com/tensorflow/tensorflow/blob/master/README.md#patching-guidelines).

Which basically explains to Build the Tensorflow pip package with the desired python version and cherrypick the chnages you need in the 2.11.0 branch. Thanks!

**Comment by YONGARYZ** [comment]
I see. 
Thanks for the comment!

**Comment by thomdixon** [comment]
> Hi,
> 
> Thanks for reporting the issue, The usual practice in Tensorflow to support any Python version is, during the time of release if that Python version is within the support life cycle as defined here https://devguide.python.org/versions/. As per the above link, during the patch release of `2.11.1` Python `3.7` was at `end-of-life `.

This is false. 

1. 2.11.1 was released on 3/20: https://github.com/tensorflow/tensorflow/releases/tag/v2.11.1
2. The CVE was published 3/24: https://github.com/advisories/GHSA-gw97-ff7c-9v96
3. This issue was reported to you on 3/30. 
4. Python 3.7 became EoL on 6/27 (as per your own link).

Folks were left vulnerable for three months while this issue went unacknowledged *until* Python 3.7 became EoL and that was used as an excuse not to fix it. Fair enough, but no need to retcon when the timeline is public.

**Comment by sachinprasadhs** [comment]
Apologies for the confusion.

It is challenging to follow the strict timeline as mentioned in the above comment.

2.11.1 (3/20) was released in the same timeline as 2.12.0(3/23) and both followed the same python support window as mentioned below for 2.12.0. 

<h2 id="tested_build_configurations" data-text="Tested build configurations" role="presentation" style="box-sizing: inherit; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-weight: ; font-stretch: ; font-size: ; line-height: ; font-family: ; font-optical-sizing: ; font-kerning: ; font-feature-settings: ; font-variation-settings: ; letter-spacing: normal; margin: var(--devsite-h2-margin); overflow: hidden; text-overflow: ellipsis; border-bottom: var(--devsite-h2-border,var(--devsite-secondary-border)); padding: var(--devsite-h2-padding); margin-inline-end: -40px; padding-inline-end: 40px; color: rgb(32, 33, 36); orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="devsite-heading" role="heading" aria-level="2" style="box-sizing: inherit;">Tested build configurations</span><button type="button" class="devsite-heading-link button-flat material-icons" aria-label="Copy link to this section: Tested build configurations" data-title="Copy link to this section: Tested build configurations" data-id="tested_build_configurations" style="box-sizing: border-box; appearance: none; background: 0px center; border: 0px; border-radius: var(--devsite-button-border-radius,2px); box-shadow: none; color: var(--devsite-icon-color,var(--devsite-secondary-text-color)); cursor: pointer; display: inline-block; font-style: normal; font-variant-ligatures: ; font-variant-caps: ; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-weight: normal; font-stretch: ; font-size: 24px; font-family: &quot;Material Icons&quot;; font-optical-sizing: ; font-kerning: ; font-feature-settings: &quot;liga&quot;; font-variation-settings: ; height: 24px; letter-spacing: normal; line-height: 1; margin: var(--devsite-button-margin,0); margin-inline-end: var(--devsite-button-margin-x-end); max-width: var(--devsite-button-max-width,none); min-width: 36px; outline: 0px; overflow: hidden; padding: 0px 8px; text-align: center; text-decoration: none; text-overflow: ellipsis; text-transform: none; transition: background-color 0.2s ease 0s, border 0.2s ease 0s, box-shadow 0.2s ease 0s; vertical-align: middle; white-space: nowrap; width: var(--devsite-button-width,auto); overflow-wrap: normal; direction: ltr; -webkit-font-smoothing: antialiased; opacity: 0;"></button></h2><h3 id="cpu" data-text="CPU" role="presentation" style="box-sizing: inherit; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-weight: ; font-stretch: ; font-size: ; line-height: ; font-family: ; font-optical-sizing: ; font-kerning: ; font-feature-settings: ; font-variation-settings: ; letter-spacing: normal; margin-top: 0px; margin-right: ; margin-bottom: ; margin-left: ; overflow: hidden; text-overflow: ellipsis; margin-inline-end: -40px; padding-inline-end: 40px; color: rgb(32, 33, 36); orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="devsite-heading" role="heading" aria-level="3" style="box-sizing: inherit;">CPU</span><button type="button" class="devsite-heading-link button-flat material-icons" aria-label="Copy link to this section: CPU" data-title="Copy link to this section: CPU" data-id="cpu" style="box-sizing: border-box; appearance: none; background: 0px center; border: 0px; border-radius: var(--devsite-button-border-radius,2px); box-shadow: none; color: var(--devsite-icon-color,var(--devsite-secondary-text-color)); cursor: pointer; display: inline-block; font-style: normal; font-variant-ligatures: ; font-variant-caps: ; font-variant-numeric: ; font-variant-east-asian: ; font-variant-alternates: ; font-weight: normal; font-stretch: ; font-size: 24px; font-family: &quot;Material Icons&quot;; font-optical-sizing: ; font-kerning: ; font-feature-settings: &quot;liga&quot;; font-variation-settings: ; height: 24px; letter-spacing: normal; line-height: 1; margin: var(--devsite-button-margin,0); margin-inline-end: var(--devsite-button-margin-x-end); max-width: var(--devsite-button-max-width,none); min-width: 36px; outline: 0px; overflow: hidden; padding: 0px 8px; text-align: center; text-decoration: none; text-overflow: ellipsis; text-transform: none; transition: background-color 0.2s ease 0s, border 0.2s ease 0s, box-shadow 0.2s ease 0s; vertical-align: middle; white-space: nowrap; width: var(--devsite-button-width,auto); overflow-wrap: normal; direction: ltr; -webkit-font-smoothing: antialiased; opacity: 0;"></button></h3><div class="devsite-table-wrapper" style="box-sizing: inherit; margin: var(--devsite-table-margin,16px 0); padding: 0px; overflow: auto; color: rgb(32, 33, 36); font-family: Roboto, &quot;Noto Sans&quot;, &quot;Noto Sans JP&quot;, &quot;Noto Sans KR&quot;, &quot;Noto Naskh Arabic&quot;, &quot;Noto Sans Thai&quot;, &quot;Noto Sans Hebrew&quot;, &quot;Noto Sans Bengali&quot;, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Version | Python version | Compiler | Build tools
-- | -- | -- | --
tensorflow-2.12.0 | 3.8-3.11 | MSVC 2019 | Bazel 5.3.0

</div>

However, as I mentioned above, you can follow [Patching guidelines](https://github.com/tensorflow/tensorflow/blob/master/README.md#patching-guidelines) for any patch specific to the python version you need.
